lib/Logger.js

```diff
@@ -32,20 +36,28 @@ class Logger {
      *                                      the 'level' field is not accounted
      *                                      for)
      *
-     * @param {string} name  - The name of the Logger. It can be found later on
-     *                         in the log entries.
      *
      * @see [Bunyan's documentation]{@link
      * https://github.com/trentm/node-bunyan/blob/master/README.md#streams} for
      * a more detailed description of the streams array configuration.
      *
      * @returns {undefined}
      */
-    constructor(config, name) {
-        this.config = config;
+    constructor(name, config) {
```

Comment:
Related to my comment on api.js: If you invert those, then we go back to the issue where:
 * It is harder to pre-bind a common (per-`api`) config into all the loggers spawned by that API